src/App.tsx

```diff
@@ -1,32 +1,57 @@
-import { useEffect, useState } from 'react';
-import './App.css';
-
-const apiKey = process.env.REACT_APP_OMDB_API_KEY;
-
+import { useContext, useState } from "react";
+import "./App.css";
+import MovieInfo from "./components/MovieInfo";
+import { DataContext } from "./context/DataContext";
+import { ButtonClickEvent, ChangeEvent } from "./customTypes";
+import ErrorAlert from "./components/ErrorAlert";
 function App() {
-  
-  const [data, setData] = useState([]);
-  
-  useEffect(() => { 
-
-    fetch(`http://www.omdbapi.com/?apikey=${apiKey}&s=batman`, { 
-      method: "GET"
-    })
-    .then((response: any) => response.json())
-    .then((data: any) => { 
-      setData(data.Search || []);
-    });
-  }, [])
-
+  const { data, queryParam, setQueryParam, fetchError } =
+    useContext(DataContext);
+  const [searchValue, setSearchValue] = useState<string>("");
+  const handleSearchInputChange = (e: ChangeEvent) => {
+    setSearchValue(e.target.value);
+  };
```

Comment:
@gabrielzepeda1 let's update this part as well, please.

```suggestion
  const handleSearchInputChange = (e: ChangeEvent) => {
  const { value } = e.target;
    setSearchValue(value);
  };
```

Reply:
When you perform code review, this is called **nitpick** @gabrielzepeda1; you can find more info about it, right here:
https://conventionalcomments.org/

It is basically a personal preference. 😃